overhaul software-info

owaddell opened the pull request:
This pull request obsoletes some entries in software-info that used to be baked into the executable in swish-win and generalizes the `software-revision` mechanism to track more information about other components.

@burgerrg, I'm still mulling over the last few commits, but would appreciate your input.

There are a few other things we might consider for the current round of testing improvements:
* `swish-test` should be writing the expected `.mo` files with some sort of "incomplete" mark so we report when something went completely sideways
* `swish-test` should have an option for generating a report from existing `.mo` files without running the tests.
* Chris suggested we consider flipping the `.mo` files over to JSON.

burgerrg commented:
I didn't know about the `profile` macro. Nice work using it!

burgerrg commented on doc/swish/app.tex:
```diff
@@ -345,6 +345,89 @@ \subsection{Configuration}
+\code{(software-info)} object.
+If \var{value} is omitted, this procedure returns the value, if any,
+associated with \var{key}, or else \code{\#f}.
+If \var{key} and \var{value} are omitted, this procedure returns the value,
```
see comment above about the trailing comma

burgerrg commented on doc/swish/app.tex:
```diff
@@ -345,6 +345,89 @@ \subsection{Configuration}
+\code{(software-info)} object.
+If \var{value} is omitted, this procedure returns the value, if any,
+associated with \var{key}, or else \code{\#f}.
+If \var{key} and \var{value} are omitted, this procedure returns the value,
```
see comment above about the trailing comma

owaddell commented:
@burgerrg I've made a few changes since 39df5c1, could you take another look?

- flipped cli-specs keys from strings to symbols
- documented inherited parameters
- folded Scheme Testing Library into swish.pdf
- added some tests
- added more mat meta-data
- moved the profile.ss changes to another branch
- changed the HTML mat-report a bit more

owaddell commented on doc/swish/testing.tex:
```diff
@@ -114,10 +120,11 @@
 The \code{run-mats-to-file} procedure executes all mats and writes
 the results into the file specified by the string \var{filename}. If
 the files exists, its contents are overwritten. The file format is a
```
Oops, missed this again in the last round of doc fixes. Good thing we have the resolve conversation workflow.

owaddell commented:
@burgerrg thanks for all the reviewing! I aim to merge it this afternoon.
